class.jetpack-data.php

```diff
@@ -26,12 +31,43 @@ public static function get_access_token( $user_id = false ) {
 			if ( $user_id != $token_chunks[2] ) {
 				return false;
 			}
-			$token = "{$token_chunks[0]}.{$token_chunks[1]}";
+			$tokens = array( "{$token_chunks[0]}.{$token_chunks[1]}" );
 		} else {
+			$tokens = Jetpack_Constants::is_defined( 'JETPACK_BLOG_TOKEN' ) && ';stored;' !== $token_key
+				? explode( ',', Jetpack_Constants::get_constant( 'JETPACK_BLOG_TOKEN' ) )
+				: array();
+
 			$token = Jetpack_Options::get_option( 'blog_token' );
-			if ( empty( $token ) ) {
+			if ( empty( $token ) && empty( $tokens ) ) {
+				return false;
+			}
+
+			$tokens[] = $token;
+		}
+
+		if ( false === $token_key ) {
+			// Use first token.
+			$token = $tokens[0];
+		} elseif ( ';stored;' === $token_key ) {
+			// Use first stored token.
+			$token = $tokens[0]; // $tokens only contains stored tokens because of earlier check.
+		} else {
+			// Use the token matching $token_key or false if none.
+			// Ensure we check the full key.
+			$token_check = rtrim( $token_key, '.' ) . '.';
+
+			$valid_token = false;
+			foreach ( $tokens as $token ) {
+				if ( hash_equals( substr( $token, 0, strlen( $token_check ) ), $token_check ) ) {
```

Comment:
What's the value of `$token_check` when this line fails?

Reply:
I can't reproduce anymore, on the same site as earlier :\

Reply:
Hm. Was it breaking for all tests (automatic, all manual ones), or just certain tests?

Reply:
It was breaking when I was doing requests to `/sites/$site` via the developer console.

Reply:
I think this was fixed in:
* b8ed3ad9020772f2664e6f51ef07804d3624eb98: never add `false` to the list of `$possible_tokens`
* 820dca9513fee0beacb04534ad232931d547c00d: allow `JETPACK_BLOG_TOKEN` to define a "Normal" (non-"Special") token. (Normal ones are… normal, Special ones are Atomic tokens, for example: the reason we're doing this in the first place :) The original code did not anticipate normal tokens in the constant, because it conflated Normal/Special v. Defined/Stored.